version: '3'
services:
  dp-postgres:
    image: postgres:16.3
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: db-tgbot
      POSTGRES_PASSWORD: tgbot
      POSTGRES_DB: gino-tgbot
    networks:
      - postgres-network
    volumes:
      - ./db-data/:/var/lib/postgresql/data/
    restart: on-failure

  py-test-bot:
    build: .
    command: python3.10 app.py
    env_file:
      - .env
    restart: always
    networks:
      - postgres-network
    depends_on:
      - dp-postgres

networks:
  postgres-network:
    driver: bridge